### ISSUE TITLE
Create a more fleshed out notion of SetUp phases in server layer

### ISSUE THREAD

**Issue by jkomoros** [jkomoros/boardgame#754]
Related to #752 and #753 .

Currently set up has to be done before the core engine actually starts the game logic. That requires variants to be done in the core logic, and also is pretty inflexible.

Instead, it should be done in a special Phase in the main game logic. This also allows games do to multiple "rounds" easily. 

To do this, we have to have machinery for known-semantics of phases that tell the server app, "at this stage, please show this type of machinery". For example, if you're in "player selection" phase, you might show a pop-over dialog with a different renderer. And if you haven't yet seen the core game renderer, then you might not even pop over to it from the game creation screen yet, showing the dialog over top of hte game list.

This requires the server to know  which phases allow player selection, variant selection, etc. Maybe a new method that takes a phase number and returns whether it's in player selection?

**Comment by jkomoros** [comment]
- [ ] Variants should use this machinery (#753)
- [ ] Player selection should use this machinery (#752)

**Comment by jkomoros** [comment]
Solved differently than originally envisioned, but the goals are met.

Original plan: special server-layer phase metadata for setup phases.
Actual solution: move legality IS the detection mechanism. No special phase metadata needed.

- Player selection (#752): ✅ Solved via SelectTeam/SelectRole/SelectColor + gathering phase
- When players may join (#755): ✅ Solved via Seat behaviors + DefaultRoundSetup + gathering panel
- Variants (#753): Left open — variants stay fixed at creation time. Mutable variants were evaluated and deferred (architecturally complex for marginal benefit; see GAME_SETUP_DESIGN.md).

The gathering system achieves everything this issue wanted — lobby UI, flexible pre-game setup, multi-round re-gathering — without any special server-layer phase machinery. The server doesn't need to know about "setup phases" because the client auto-detects gathering state from move legality.
